ansible_collections/arista/avd/roles/eos_designs/python_modules/base/avdstructuredconfig.py

```diff
@@ -368,14 +368,18 @@ def queue_monitor_length(self) -> dict | None:
         return queue_monitor_length_dict
 
     @cached_property
-    def name_server(self) -> dict | None:
+    def ip_name_servers(self) -> list | None:
         """
-        name_server set based on name_servers data-model and mgmt_interface_vrf
+        ip_name_servers set based on name_servers data-model and mgmt_interface_vrf
         """
-        if (name_servers := get(self._hostvars, "name_servers")) is not None:
-            return {"source": {"vrf": self._mgmt_interface_vrf}, "nodes": name_servers}
-
-        return None
+        ip_name_servers = [
+            {
+                "ip_address": name_server,
+                "vrf": self._mgmt_interface_vrf,
+            }
+            for name_server in get(self._hostvars, "name_servers", default=[])
+        ]
+        return ip_name_servers or None
```

Comment:
For now, just use the if check and return like elsewhere. Or should we change in general to this?

Reply:
this was a suggested refactor from sourcery - I can change it back no problem :)